cudapoa/src/cudapoa_batch.hpp

```diff
@@ -162,12 +604,19 @@ class CudapoaBatch : public Batch
     bool banded_alignment_;
 
     // Pointer of a seperate class BatchBlock that implements details on calculating and allocating the memory for each batch
-    std::unique_ptr<BatchBlock> batch_block_;
+    std::unique_ptr<BatchBlock<ScoreT>> batch_block_;
 
     // Maximum POAs to process in batch.
     int32_t max_poas_ = 0;
+
+public:
+    // Static batch count used to generate batch IDs.
+    static int32_t batches;
 };
 
+template <typename ScoreT>
```

Comment:
this is just a counter for how many cudapoa batch objects are created. this doesn't need to be templated by `ScoreT`. it's more like an ID of the batch, not the nodes in the graph

Reply:
right, but it is defined inside CudapoaBatch class (a couple of lines above it) which this class is now templated. So can't call it like `CudapoaBatch::batches = 0;` you want to take it out from the class, or just leave it as it is.

Reply:
Ah that's right! My bad, I got confused. Yes let's leave this as it is.